zipkin-cassandra-core/src/main/java/org/twitter/zipkin/storage/cassandra/Repository.java

```diff
@@ -496,7 +516,7 @@ private String debugInsertServiceName(String serviceName, int ttl) {
 
                 BoundStatement bound = selectSpanNames.bind()
                         .setString("service_name", serviceName)
-                        .setList("bucket", ALL_BUCKETS);
+                        .setInt("bucket", 0);
```

Comment:
it won't be (afaik) needed.

no service_span combination will be written more than a handful of times each hour (for each collector jvm running).

from a write per span stored to just a few per hour i do think removes the need for bucketing.